mlops/parallelm/mlops/mlops.py

```diff
@@ -932,6 +933,52 @@ def load_time_capture(self, input_file):
         from parallelm.mlops.time_capture.time_capture import TimeCapture
         return TimeCapture(input_file)
 
+    def attach(self,
+               mlapp_id,
+               mlops_server=Constants.MLOPS_DEFAULT_HOST,
+               mlops_port=Constants.MLOPS_DEFAULT_PORT,
+               user=Constants.MLOPS_DEFAULT_USER,
+               password=None):
+        """
+        Attach to a running ION and run in its context.
+        Side effect: sets up mlops_context
+        :param mlapp_id: the id of the ION to connect to
+        :param mlops_server: the host to connect to
+        :param mlops_port: the port MLOps is using
+        :param user: user name to use for connection
+        :param password: password to use for authentication, if None (default) then prompt for password
+        :return:
+    `
+        Note: Attach only works for pure python code
+        """
+        self._logger.info("Connecting to mlops: {} {}: {} user: {} pass: {}".format(
+            mlops_server, Constants.ION_LITERAL, mlapp_id, user, password))
+
+        # Connecting directly the server
+        rest_helper = MlOpsRestFactory().get_rest_helper(MLOpsMode.ATTACH, mlops_server, mlops_port, None)
+        token = rest_helper.login(user, password)
+
+        # Setting the environment for mlops
+        ci = ConfigInfo()
+        ci.token = token
+        ci.zk_host = None
+        ci.mlops_port = str(mlops_port)  # Constants.MLOPS_DEFAULT_PORT
+        ci.mlops_server = mlops_server
+        ci.ion_id = mlapp_id
+        ci.mlops_mode = MLOpsMode.ATTACH
+        ci.output_channel_type = OutputChannel.PYTHON
+
+        # TODO: for now assume node "0" - allow providing the node id or just become any node
+        ci.ion_node_id = "0"
+        ci.pipeline_id = "0"
```

Comment:
given this is just a string constant for all practical purposes, this should be added to the constants file and maybe make it "NA"? or "not-needed" or something to make it clear that its not being actually used

Reply:
Looks like it is used. When I change it I get:
"Current node id: [NA] is not detected in ML App" from mlops context. I will keep it for now not to do deeper changes but this requires future changes.

Reply:
that is odd - this is a bug in our code then - we shouldnt be using the node-id when we attach? can we fix it correctly instead of hardcoding a bad node id?

Reply:
I think it is OK. We re-use the python output channel for the attach mode. and we plug in the node id for that. it's not a real node Id. 
Not the cleanest way, but I'm not sure it worth messing with for now.